### PR TITLE
Setup initial test data when running Python unit tests

### PR DESCRIPTION
**Pull request by chosak**
This PR tries to setup required initial test data before running the Python unit tests, by implementing a custom Django TestRunner class.

## Additions

- Added new `cfgov/cfgov/test.py` file that contains a `cfgov.test.TestDataTestRunner` class.

## Removals

- Removed check against `settings.DEBUG` in `cfgov/scripts/initial_data.py` as this seemed like an unnecessary check vs. just not running the script. Is there anywhere this script is ever run in production?

## Changes

- Python testing using `manage.py test` now uses the custom test runner, which does these steps before running the tests:
 1. Checks to see if the Wagtail [initial_data](https://github.com/torchbox/wagtail/blob/v1.5.3/wagtail/wagtailcore/migrations/0002_initial_data.py) migration has been run (it normally is when running `tox` or `manage.py test`, but not when running `tox -e fast` which skips all migrations), and, if not, runs it to create the Wagtail root page and related models.
 1. Runs our `cfgov/scripts/initial_data.py` to setup initial pages like home, Events, and Archive.
 1. Runs our `cfgov/scripts/initial_test_data.py` to setup specific testing data.

## Testing

- Run `tox` and `tox -e fast` and notice that the initial test data scripts are run before the tests are.

## Review

- @cfpb/cfgov-backends 

## Todos

- If this approach makes sense, we could consider having the Django unit test runner also kick off the Protractor browser tests, so that those can run against the test database.

## Checklist

* [X] Changes are limited to a single goal (no scope creep)
* [X] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [X] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by richaagarwal** [comment]
👍

**Comment by richaagarwal** [comment]
I've merged this PR into my branch `rewrite-wagtail-tests` where it is useful. Should we leave this PR open as a stand-alone or wait until my PR is merged in?

File: cfgov/scripts/initial_data.py [hunk 1]
```diff
@@ -9,66 +9,65 @@
 
 
 def run():
-    if settings.DEBUG:
-        print 'Running script \'scripts.initial_data\' ...'
-        admin_user = None
-        site_root = None
-        events = None
+    print 'Running script \'scripts.initial_data\' ...'
```
Comment (willbarton) [inline cfgov/scripts/initial_data.py]: Can we `from __future__ import print_function` and use `print()` here, since we're changing it?

Comment (chosak) [inline cfgov/scripts/initial_data.py]: Thanks, good suggestion. Implemented in 041e15c.

**Comment by chosak** [comment]
Per discussion with @richaagarwal, going to merge this in to keep PR small and unblock other work that might depend upon this test data.

**Pull request merged**